### ISSUE TITLE
File selection is invisible at Commit dialog, v3.00.00 beta1 

### ISSUE THREAD

**Issue by zlolik** [gitextensions/gitextensions#5499]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
File selection is invisible at Commit dialog, v3.00.00 beta1 

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
Just open Commit dialog and click on one the changed file

**What is the expected behavior?**
Selected file has dark-blue background.

**Environment you encounter the issue:**

- GitExtensions version: v3.00.00 beta1 
- GIT version: 2.18
- OS version: Win7_x64
- .NET version: 4.7.03062

**Did this work in previous version of GitExtensions (which)?**
2.52

![default](https://user-images.githubusercontent.com/8332364/46331802-37134e80-c622-11e8-98bf-da9f1a2e093c.png)

**Comment by RussKie** [comment]
It feels related to #5379. It is probably classic theme specific....
Someone on W7 needs to debug it

**Comment by t-jet** [comment]
I can confirm this issue. Also, it's reproducible on Win10 with classic theme.
Also, I found root cause, but can't find enough free time slot to fix it or, even, report it for a month already :( 

Source file is **GitUI\UserControls\FileStatusList.cs**

Error introduced by [this commit](https://github.com/gitextensions/gitextensions/commit/d0e1b0c42c60ae52e6544c15ed962549fef8fad8). 

Logic of drawing selection background and focus rectangle was removed in line 204:

![removed_background_fill](https://user-images.githubusercontent.com/8374961/46498235-82497f00-c825-11e8-810e-b052c4594e06.png)

There are two things to do:

1. Restore beckground and focus rectangle drawing.
2. Check if item selected and use right colors while drawing text.

**Comment by RussKie** [comment]
Awesome, thank you for looking into it.

**Comment by rodionkvashnin** [comment]
I can confirm that this problem exists in the GitExtensions 3.00.00.4433 (Windows 7 64-bit, Windows Classic Theme).

**Comment by RussKie** [comment]
I've been observing this issue one a regular basis on my work laptop.
- Git Extensions 3.0.1.4933
- Build a269358b22a5dc1ea36928bb87f9e535de76a058
- Git 2.19.2.windows.1
- Microsoft Windows NT 10.0.17763.0
- .NET Framework 4.7.3260.0
- DPI 192dpi (200% scaling)

I can't debug it there, and not sure how to repro the problem on my personal laptop.

**Comment by firefly4321** [comment]
I am on Window 7. I can reliably reproduce the problem by turning on "Adjust for Best Performance" under Performant Options->Visual Effects.

The problem goes away when I turn on "Adjust for best appearance".

**Comment by Sharuru** [comment]
Once you turn on the 'Adjust for best appearance' in Windows 7, it will enable the Aero theme makes the problem away.

But for users who keen on Classic theme, this workaround does not work well. 

Maybe there are some 'magic things' on Windows about DPIs, themes and so on :P

**Comment by RussKie** [comment]
@t-jet suggested a possible fix in https://github.com/gitextensions/gitextensions/issues/5499#issuecomment-427144570
I suppose we need a volunteer to try it and see if it works.

**Comment by mast-eu** [comment]
> @t-jet suggested a possible fix in [#5499 (comment)](https://github.com/gitextensions/gitextensions/issues/5499#issuecomment-427144570)
> I suppose we need a volunteer to try it and see if it works.

I tested it on Win7 with classic theme. The issue still applies on 3.1-RC1. Unfortunately, the suggested fix is no longer applicable, because it is based on a quite old commit from Feb 2018. There have been major changes since then due to the work on GE version 3.

Anyway, I'm in a position to debug it - sooner or later. Assign issue to myself to remember.

**Comment by snelltheta** [comment]
Issue still applies in Git Extensions 3.1.1.6049 on Windows 7.
Can be reproduced by disabling "Use visual styles on windows and buttons" in the Performance Options of Windows 7.

**Comment by gerhardol** [comment]
Changing the visual appearance is the solution, a change for Win7 Classic theme is unfortunately not likely, sorry.
(@mast-eu is busy with real life and is not expected to look at this for some time - will he have Win7 when coming back?)

**Comment by mast-eu** [comment]
> @mast-eu is busy with real life and is not expected to look at this for some time

True. Real life has taken a large amount of my previously free time. And within the time I can currently dedicate to GE, this issue has a rather low priority.

>   will he have Win7 when coming back?

Yes. I still have a Win7 dev machine at home and no plans to upgrade the OS.

**Comment by RussKie** [comment]
I observed it randomly happening on w10 as well in the Diff tab but haven't
been able to repro it consistently.... maybe related.

**Comment by snelltheta** [comment]
Still a problem in 
- Git Extensions 3.3.0.7719
- Build e52f5178e8d3c9899efbe6fae66026865afa3fe1
- Git 2.24.1.windows.2
- Microsoft Windows NT 6.1.7601 Service Pack 1
- .NET Framework 4.7.3416.0
- DPI 96dpi (no scaling)

**Comment by EleonoreMizo** [comment]
I can confirm file selection is still invisible with Git Extensions 3.3.0.7719 in Windows 7 classic

**Comment by vrubleg** [comment]
This simple change fixes the problem:
```
diff --git a/GitUI/UserControls/FileStatusList.cs b/GitUI/UserControls/FileStatusList.cs
index 6233b45f8..7ee8476af 100644
--- a/GitUI/UserControls/FileStatusList.cs
+++ b/GitUI/UserControls/FileStatusList.cs
@@ -1400,6 +1400,11 @@ private void FileStatusListView_DrawSubItem(object sender, DrawListViewSubItemEv
 
             var (image, prefix, text, suffix, prefixTextStartX, _, textMaxWidth) = FormatListViewItem(item, formatter, item.Bounds.Width);
 
+            if (e.Item.Selected)
+            {
+                e.Graphics.FillRectangle(SystemBrushes.Highlight, e.Bounds);
+            }
+
             if (image != null)
             {
                 e.Graphics.DrawImageUnscaled(image, item.Position.X, item.Position.Y);
@@ -1411,18 +1416,18 @@ private void FileStatusListView_DrawSubItem(object sender, DrawListViewSubItemEv
 
                 if (!string.IsNullOrEmpty(prefix))
                 {
-                    DrawString(textRect, prefix, SystemColors.GrayText);
+                    DrawString(textRect, prefix, e.Item.Selected ? SystemColors.HighlightText : SystemColors.GrayText);
                     var prefixSize = formatter.MeasureString(prefix);
                     textRect.Offset(prefixSize.Width, 0);
                 }
 
-                DrawString(textRect, text, SystemColors.ControlText);
+                DrawString(textRect, text, e.Item.Selected ? SystemColors.HighlightText : SystemColors.ControlText);
 
                 if (!string.IsNullOrEmpty(suffix))
                 {
                     var textSize = formatter.MeasureString(text);
                     textRect.Offset(textSize.Width, 0);
-                    DrawString(textRect, suffix, SystemColors.GrayText);
+                    DrawString(textRect, suffix, e.Item.Selected ? SystemColors.HighlightText : SystemColors.GrayText);
                 }
             }
```
Tested it on Windows 7 with Classic Theme and on Windows 10 with default theme. The look and feel of the selected row in this list will be the same as in the commits list. It looks better when both selected rows in different lists are in the same style.

If you are OK with this change, I'll make a PR.

**Comment by gerhardol** [comment]
> do you plan to make another 3.x version (with some bug fixes) before 4.x?

No 3.x is planned - a release will basically stop @RussKie from other work for a month. Not stability, but being involved in adopting changes that some users do not like. (Like this: While some dislike using Windows colors, some will not like the change either.)

We welcome users to run the portable builds in master. Maybe there could be alpha releases with release notes?

**Comment by RussKie** [comment]
> do you plan to make another 3.x version (with some bug fixes) before 4.x?

Yes, I've been wondering the same Q :) I think we've done quite a bit of work since 3.4.3, so it is not totally unreasonable to ask this.
There are a few changes that I would consider important to complete (e.g. the left panel related work) before we can really consider this.
The "4" number is arbitrary too, so we can change it back to be "3.5" :)



> a release will basically stop @RussKie from other work for a month.

Yes, each release is an involved process no matter how much we try to automate it.
